dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -279,24 +271,22 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
 
         TagList tags = new() { { MeasurementFunctionTagName, this.Name } };
         long startingTimestamp = Stopwatch.GetTimestamp();
+
         try
         {
-            IAsyncEnumerator<TResult> enumerator;
+            IAsyncEnumerator<TResult>? enumerator = null;
             try
             {
                 // Quick check for cancellation after logging about function start but before doing any real work.
                 cancellationToken.ThrowIfCancellationRequested();
 
-                // Invoke pre-invocation filter. If it requests cancellation, throw.
-                var invokingContext = kernel.OnFunctionInvokingFilter(this, arguments);
-
-                if (invokingContext?.Cancel is true)
+                var invocationContext = await kernel.OnFunctionInvocationAsync(this, arguments, (context) =>
                 {
-                    throw new OperationCanceledException("A function filter requested cancellation before function invocation.");
-                }
+                    // Invoke the function and get its streaming enumerator.
+                    enumerator = this.InvokeStreamingCoreAsync<TResult>(kernel, context.Arguments, cancellationToken).GetAsyncEnumerator(cancellationToken);
```

Comment:
That's one of the limitations I described in ADR for streaming scenarios. I'm not sure how to override streamed response and handle exceptions during streaming yet. Will think about it.

Reply:
>  I'm not sure how to override streamed response and handle exceptions during streaming yet.

We can't just let the filter supply its own IAsyncEnumerable? It could just wrap the original, and unless it eats exceptions (which would be up to it as a filter), then any exceptions that emerge from the underlying enumerable would also surface from the wrapper. Basically the filter creates a decorator.

Reply:
That's a great idea. I implemented it and it allows to modify streaming response or catch exceptions during streaming. I also provided samples with these scenarios. Thanks a lot!